include/oneapi/dpl/internal/dynamic_selection_impl/round_robin_policy.h

```diff
@@ -0,0 +1,156 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_ROUND_ROBIN_POLICY_IMPL_H
+#define _ONEDPL_ROUND_ROBIN_POLICY_IMPL_H
+
+#include <atomic>
+#include <type_traits>
+#include <vector>
+#include <memory>
+#include <exception>
+#include <limits>
+#include "oneapi/dpl/internal/dynamic_selection_impl/scoring_policy_defs.h"
+#if _DS_BACKEND_SYCL != 0
+    #include "oneapi/dpl/internal/dynamic_selection_impl/sycl_backend.h"
+#endif
+
+namespace oneapi {
+namespace dpl{
+namespace experimental{
+#if _DS_BACKEND_SYCL != 0
+  template <typename Backend=sycl_backend>
```

Comment:
I am concerned about "one definition rule". I cannot say for sure what could go wrong in practice but the code looks dangerous if we have some binaries built with and without `-fsycl`.

Let's think together what might go wrong and how we can fix that... (Eliminate the default argument at all?)

Reply:
When built with `-fsycl`, `round_robin_policy<>` means `round_robin_policy<sycl_backend>`; otherwise,  `round_robin_policy<>` is a compilation error. Does not sound like ODR violation to me :)

Reply:
Ok, we will (in the future) introduce type aliases and remove the default argument. But for now, we will keep the current implementation.